test/functional/page_objects/visualize_page.ts

```diff
@@ -59,6 +67,21 @@ export function VisualizePageProvider({ getService, getPageObjects }: FtrProvide
         .map((chart) => $(chart).findTestSubject('visTypeTitle').text().trim());
     }
 
+    public async getPromotedVisTypes() {
+      const chartTypeField = await testSubjects.find('visNewDialogGroups');
+      const $ = await chartTypeField.parseDomContent();
+      const promotedVisTypes: string[] = [];
+      $('button')
+        .toArray()
+        .map((chart) => {
```

Comment:
nit: 'map' method confuse a little. I think it's better to use `forEach` if you want to keep filtered values in parent scope (promotedVisTypes var), or use `reduce`  instead of `map`